### PR TITLE
chore(tests): convert Translator golden tests into KongClient ones

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Converts the `Translator` golden tests into `KongClient` ones. This will allow us to cover the whole `KongClient.Update` flow, making these tests more reliable as we will store the content of the config requests that got to the Admin API instead of the artificially marshaled `file.Content` that we were getting before by manually calling `deckgen.ToDeckContent` in the test.

**Which issue this PR fixes**:

It's a prerequisite for https://github.com/Kong/kubernetes-ingress-controller/issues/6076.

### CODE REVIEW

File: internal/dataplane/kong_client_golden_test.go [hunk 1]
```diff
@@ -233,20 +236,56 @@ func runTranslatorGoldenTest(t *testing.T, tc translatorGoldenTestCase) {
 	p, err := translator.NewTranslator(logger, s, "", tc.featureFlags, fakeSchemaServiceProvier{})
 	require.NoError(t, err, "failed creating translator")
 
-	// MustBuild the Kong configuration.
-	result := p.BuildKongConfig()
-	targetConfig := deckgen.ToDeckContent(context.Background(),
+	timeout := time.Second
```
Comment (programmer04) [inline internal/dataplane/kong_client_golden_test.go:239]:
```suggestion
	const timeout = time.Second
```

File: test/kongintegration/translator_golden_tests_outputs_test.go [hunk 2]
```diff
@@ -108,35 +99,47 @@ func TestTranslatorsGoldenTestsOutputs_Konnect(t *testing.T) {
...
+			err = yaml.Unmarshal(goldenTestOutput, content)
+			require.NoError(t, err)
+
+			require.Eventually(t, func() bool {
+				if err := updateStrategy.Update(ctx, sendconfig.ContentWithHash{Content: content}); err != nil {
+					t.Logf("error: %v", err)
+					return false
+				}
+				return true
+			}, timeout, tick)
```
Comment (programmer04) [inline test/kongintegration/translator_golden_tests_outputs_test.go:115]: Maybe it is better to use `require.EventuallyWithT` to not spam with some redundant logs, WDYT?

File: test/kongintegration/translator_golden_tests_outputs_test.go [hunk 3]
```diff
@@ -108,35 +99,47 @@ func TestTranslatorsGoldenTestsOutputs_Konnect(t *testing.T) {
...
-	goldenTestOutputPath string,
-	sut sendconfig.UpdateStrategy,
-) {
+func ensureGoldenTestOutputIsAccepted(ctx context.Context, t *testing.T, goldenTestOutputPath string, kongClient *kong.Client) {
 	goldenTestOutput, err := os.ReadFile(goldenTestOutputPath)
 	require.NoError(t, err)
 
-	content := &file.Content{}
-	err = yaml.Unmarshal(goldenTestOutput, content)
+	cfg := map[string]interface{}{}
```
Comment (programmer04) [inline test/kongintegration/translator_golden_tests_outputs_test.go:124]:
```suggestion
	cfg := map[string]any{}
```
it should be a linter to force it across the codebase. I need to check 😉

File: test/kongintegration/translator_golden_tests_outputs_test.go [hunk 4]
```diff
@@ -108,35 +99,47 @@ func TestTranslatorsGoldenTestsOutputs_Konnect(t *testing.T) {
...
+	err = yaml.Unmarshal(goldenTestOutput, &cfg)
+	require.NoError(t, err)
+
+	cfgAsJSON, err := json.Marshal(cfg)
 	require.NoError(t, err)
 
 	require.Eventually(t, func() bool {
-		if err := sut.Update(ctx, sendconfig.ContentWithHash{Content: content}); err != nil {
-			t.Logf("error: %v", err)
+		resp, err := kongClient.ReloadDeclarativeRawConfig(ctx, bytes.NewReader(cfgAsJSON), true, true)
```
Comment (programmer04) [inline test/kongintegration/translator_golden_tests_outputs_test.go:132]: `require.EventuallyWithT`?

File: test/mocks/admin_api_handler.go [hunk 5]
```diff
@@ -440,3 +444,21 @@ const defaultDBLessStatusResponseWithoutConfigurationHash = `{
 	  "connections_active": 3
 	}
 }`
+
+// convertReceivedJSONConfigIntoGetConfigResponse converts the received JSON config into a response for a `GET /config` request.
+// That's the way Kong Gateway behaves and to satisfy kong.Client expectations we need to do the same in the mock server.
+func convertReceivedJSONConfigIntoGetConfigResponse(t *testing.T, jsonConfig []byte) []byte {
+	cfg := map[string]interface{}{}
```
Comment (programmer04) [inline test/mocks/admin_api_handler.go:451]:
```suggestion
	cfg := map[string]any{}
```

**Review by programmer04 — COMMENTED** [review]
Just some nits

File: internal/dataplane/kong_client_golden_test.go [hunk 6]
```diff
@@ -203,21 +206,21 @@ func unmarshalSettingsFile(t *testing.T, path string) translatorSettings {
...
+// file for a specific combination of feature flags.
+type kongClientGoldenTestCase struct {
 	k8sConfigFile string
 	goldenFile    string
 	featureFlags  translator.FeatureFlags
 }
 
-func runTranslatorGoldenTest(t *testing.T, tc translatorGoldenTestCase) {
+func runKongClientGoldenTest(t *testing.T, tc kongClientGoldenTestCase) {
 	logger := zapr.NewLogger(zap.NewNop())
```
[the remark this answers is not in the document]
Comment (pmalek) [reply internal/dataplane/kong_client_golden_test.go]: That's nice but what I meant (and communicated poorly 😢 ) is to log those e.g. feature flags (or include them in the testcase name)

[the remark this answers is not in the document]
Comment (czeslavo) [reply internal/dataplane/kong_client_golden_test.go]: No worried :) Makes sense, added logs dumping feature flags as well as input and golden files paths.

File: internal/dataplane/kong_client_golden_test.go [hunk 7]
```diff
@@ -131,24 +134,24 @@ func pruneTestCaseDirectory(t *testing.T, path string) {
...
 
-// resolveSetsOfTranslatorSettingsForTestCaseDir returns a slice of translatorSettings, each of which represents a combination of
+// resolveSetsOfSettingsForTestCaseDir returns a slice of testCaseSettings, each of which represents a combination of
 // feature flags and Kong version.
 // The function iterates over a test case directory containing zero or more files named "<name>_settings.yaml".
-// If it doesn't find any settings files, it returns a single translatorSettings with default feature flags and Kong version.
-func resolveSetsOfTranslatorSettingsForTestCaseDir(t *testing.T, path string) []translatorSettings {
+// If it doesn't find any settings files, it returns a single testCaseSettings with default feature flags and Kong version.
+func resolveSetsOfSettingsForTestCaseDir(t *testing.T, path string) []testCaseSettings {
 	// Iterate over all files in the directory and look for settings files.
```
Comment (pmalek) [inline internal/dataplane/kong_client_golden_test.go:142]: nit:

```suggestion
func resolveSetsOfSettingsForTestCaseDir(t *testing.T, path string) []testCaseSettings {
	t.Helper()

	// Iterate over all files in the directory and look for settings files.
```

File: internal/dataplane/kong_client_golden_test.go [hunk 8]
```diff
@@ -203,22 +206,29 @@ func unmarshalSettingsFile(t *testing.T, path string) translatorSettings {
...
 }
 
-func runTranslatorGoldenTest(t *testing.T, tc translatorGoldenTestCase) {
-	logger := zapr.NewLogger(zap.NewNop())
+// runKongClientGoldenTest runs a single golden test case for the KongClient.
+func runKongClientGoldenTest(t *testing.T, tc kongClientGoldenTestCase) {
+	t.Helper()
+
+	t.Logf("Running test case with input file %s and golden file %s", tc.k8sConfigFile, tc.goldenFile)
+	t.Logf("Feature flags: %+v", tc.featureFlags)
```
Comment (pmalek) [inline internal/dataplane/kong_client_golden_test.go:231]: 👍

**Review by pmalek — APPROVED** [review]

**Review by pmalek — APPROVED** [review]

**Review by programmer04 — APPROVED** [review]

**Pull request merged**